### ISSUE TITLE
How can I use the this result?

### ISSUE THREAD

**Issue by bboyxu5928** [huggingface/transformers#3044]
I wan to call the next sentence prediction function on new data. 
 and this webpage tell it 
https://stackoverflow.com/questions/55111360/using-bert-for-next-sentence-prediction

 input_ids = torch.LongTensor([[31, 51, 99], [15, 5, 0]])
    input_mask = torch.LongTensor([[1, 1, 1], [1, 1, 0]])
    token_type_ids = torch.LongTensor([[0, 0, 1], [0, 1, 0]])
    config = BertConfig(vocab_size_or_config_json_file=32000, hidden_size=768,
        num_hidden_layers=12, num_attention_heads=12, intermediate_size=3072)
    model = BertForNextSentencePrediction(config)
    seq_relationship_logits = model(input_ids, token_type_ids, input_mask)

but when I run this demo and get the result ( seq_relationship_logits) ,
(tensor([[-0.0728,  0.1863],
        [ 0.3190, -0.1528]], grad_fn=<AddmmBackward>),)
how to use it as the predict result （like if sentence B follows sentence A, so the predict label is 0,else the label is 1）

**Comment by BramVanroy** [comment]
The returned values are the logits before softmax, so to get probabilities you still need to pull 'em through a softmax function, e.g.:

```python
import torch
from torch.nn.functional import softmax
from transformers import BertConfig, BertForNextSentencePrediction

input_ids = torch.LongTensor([[31, 51, 99], [15, 5, 0]])
input_mask = torch.LongTensor([[1, 1, 1], [1, 1, 0]])
token_type_ids = torch.LongTensor([[0, 0, 1], [0, 1, 0]])

config = BertConfig(vocab_size_or_config_json_file=32000,
                    hidden_size=768,
                    num_hidden_layers=12,
                    num_attention_heads=12,
                    intermediate_size=3072)
model = BertForNextSentencePrediction(config)

# a model's output is a tuple, we only need the output tensor containing
# the relationships which is the first item in the tuple
seq_relationship_logits = model(input_ids, token_type_ids, input_mask)[0]

probs = softmax(seq_relationship_logits, dim=1)

print(seq_relationship_logits)
# index 0: sequence B is a continuation of sequence A
# index 1: sequence B is a random sequence
print(probs)
```

**Comment by BramVanroy** [comment]
That answer links to an old version of the codebase. Here is an example using the current-day implementation.

```python
from torch.nn.functional import softmax
from transformers import BertForNextSentencePrediction, BertTokenizer


seq_A = 'I like cookies !'
seq_B = 'Do you like them ?'

# load pretrained model and a pretrained tokenizer
model = BertForNextSentencePrediction.from_pretrained('bert-base-cased')
tokenizer = BertTokenizer.from_pretrained('bert-base-cased')

# encode the two sequences. Particularly, make clear that they must be 
# encoded as "one" input to the model by using 'seq_B' as the 'text_pair'
encoded = tokenizer.encode_plus(seq_A, text_pair=seq_B, return_tensors='pt')
print(encoded)
# {'input_ids': tensor([[  101,   146,  1176, 18621,   106,   102,  2091,  1128,  1176,  1172, 136,   102]]),
#  'token_type_ids': tensor([[0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1]]),
#  'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]])}
# NOTE how the token_type_ids are 0 for all tokens in seq_A and 1 for seq_B, 
# this way the model knows which token belongs to which sequence

# a model's output is a tuple, we only need the output tensor containing
# the relationships which is the first item in the tuple
seq_relationship_logits = model(**encoded)[0]

# we still need softmax to convert the logits into probabilities
# index 0: sequence B is a continuation of sequence A
# index 1: sequence B is a random sequence
probs = softmax(seq_relationship_logits, dim=1)

print(seq_relationship_logits)
print(probs)
# tensor([[9.9993e-01, 6.7607e-05]], grad_fn=<SoftmaxBackward>)
# very high value for index 0: high probability of seq_B being a continuation of seq_A
# which is what we expect!
```

**Comment by bboyxu5928** [comment]
Got it,thanks a lot!
